internal/sighandler/sighandler.go

```diff
@@ -0,0 +1,46 @@
+package sighandler
+
+import (
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/logging"
+)
+
+var sigCh chan os.Signal
+var ignoreCh chan bool
+
+// Init initializes the global signal interrupt handler
+func Init(subCommand string) {
+	sigCh = make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt)
+	ignoreCh = make(chan bool, 0)
+
+	go func(subCommand string) {
+		defer close(sigCh)
+		defer close(ignoreCh)
+		ignore := false
+		for {
+			select {
+			case <-sigCh:
+				if !ignore {
+					logging.Debug("captured ctrl-c event")
+					analytics.EventWithLabel(analytics.CatCommandExit, subCommand, "interrupt")
+					analytics.WaitForAllEvents(time.Second)
+					os.Exit(1)
```

Comment:
Why are you calling `os.Exit`? Big red flag. We want to bubble up interrupts as errors.

Reply:
I did it this way, because that is the default behavior of `ctrl-c`.  It exits the programme with a non-zero exit code.  Here we just inject our analytics code before.

I could add another channel, and run the `c.execute()` function in a go function, such that I could listen for what returns first: the main path (`c.execute` or the signal handler).  That should be a little nicer.